events.go

```diff
@@ -215,22 +251,24 @@ var NullEventStream = EventStreamFunc(func(event Event) error {
 	return nil
 })
 
-// AsyncEventStream wraps an EventStream to publish events asynchronously in a
-// goroutine.
+// AsyncEventStream wraps an array of EventStreams to publish events
+// asynchronously in a goroutine
 type AsyncEventStream struct {
 	EventStream
-	events chan Event
+
+	streams []EventStream
+	events  chan Event
 }
 
 // AsyncEvents returns a new AsyncEventStream that will buffer upto 100 events
 // before applying backpressure.
-func AsyncEvents(e EventStream) *AsyncEventStream {
-	s := &AsyncEventStream{
-		EventStream: e,
-		events:      make(chan Event, 100),
+func AsyncEvents(s []EventStream) *AsyncEventStream {
```

Comment:
You could keep this as an `EventStream` and just make `[]EventStream` implement the EventStream interface:

```go
// MutliEventStream is an EventStream implementation that publishes the event to multiple EventStreams, returning any errors after publishing to all streams.
type MultiEventStream []EventStream

func (streams MultiEventStream) PublishEvent(event Event) error {
	var errors []error
	for _, s := range streams {
		errors = append(errors, s.PublishEvent(event))
	}

	if len(errors) > 0 {
		// Probably a better way to generate a MultiError here.
		return fmt.Errorf("errors publishing to event streams: %v", errors)
	}

	return nil
}
```